File: preprocessing/x_data_binded.py
import pandas as pd
X1 = pd.read_csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-#ianos-20240823_142323.csv")
X2 = pd.read_csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-#ianos-20240825_053644.csv")
X3 = pd.read_csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-ζημιές-20240823_225412.csv")
X4 = pd.read_csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-καιρός-20240823_151540.csv")

X_total = pd.concat([X1,X2,X3,X4], ignore_index=True)
X_total_complete =  X_total.drop_duplicates()

#export as csv
X_total_complete.to_csv('/media/stathis/StathisUSB/final_classification_march_9/text_data/X/x_data_binded.csv')

File: preprocessing/x_data_binded.R
q1 <- read.csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-#ianos-20240823_142323.csv")
q2 <- read.csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-#ianos-20240825_053644.csv")
q3 <- read.csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-ζημιές-20240823_225412.csv")
q4 <- read.csv("/media/stathis/StathisUSB/final_classification_march_9/text_data/X/TwiBot-καιρός-20240823_151540.csv")

q.total <- rbind(q1,q2,q3,q4)
qtotal <- unique(q.total)

write.csv(qtotal, "/media/stathis/StathisUSB/final_classification_march_9/text_data/X/X_total.csv")

#create also a sample of 150 rows
data.sample <- sample(rownames(qtotal), 150)
data.train <- qtotal[data.sample, ]
write.csv(data.train, "/home/stathis/Desktop/Research/FLOOD/Sources/X/train.csv")
